File: src/test/testdata/inspections/unsafeIterators/Predicate.PredicateIterable.java
package main;

import com.badlogic.gdx.utils.Predicate;

import java.util.Iterator;

class MyPredicate implements Predicate {
  public boolean evaluate(Object arg0) {
    return true;
  }
}

class PredicateIterableTest {

  void test() {
    Predicate.PredicateIterable pi = new Predicate.PredicateIterable(new Iterable() {
      public Iterator iterator() {
        return null;
      }
    }, new MyPredicate());

    for (Object o: <warning>pi</warning>) {
      Iterator i = <warning>pi.iterator()</warning>;
    }
  }

}

File: src/test/testdata/inspections/unsafeIterators/Predicate.PredicateIterable.kt
package main

import com.badlogic.gdx.utils.ObjectSet
import com.badlogic.gdx.utils.Predicate

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
class MyPredicateKotlin : Predicate<Any> {
  override fun evaluate(arg0: Any): Boolean {
    return true
  }
}

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
class PredicateIterableTestKotlin {

  fun test() {
    val pi = Predicate.PredicateIterable(object : Iterable<Any> {
      override operator fun iterator(): Iterator<String> {
        return <warning>ObjectSet<String>().iterator()</warning>
      }
    }, MyPredicateKotlin())

    for (o in <warning>pi</warning>) {
      val i = <warning>pi.iterator()</warning>
    }

  }

}
